### ISSUE TITLE
Double-dash `--` separator for command-line options and commands no longer works

### ISSUE THREAD

**Issue by s417-lama** [openpmix/openpmix#2980]
### What version of the PMIx Reference Library are you using? 

```console
$ git submodule status
 22dabb8e471fb110ca75a5fd3d5f12d8d6a984a3 3rd-party/openpmix (v1.1.3-3791-g22dabb8e)
 10496e38a0b54722723ec83923f6311ec82d692b 3rd-party/prrte (psrvr-v2.0.0rc1-4582-g10496e38a0)
 237ceff1a8ed996d855d69f372be9aaea44919ea config/oac (heads/main-4-g237ceff)
```

### Describe how PMIx was installed 

from a git clone with Open MPI main branch

### Please describe the system on which you are running

* Operating system/version: RHEL 8.4
* Computer hardware: Intel Xeon CPU
* Network type: InfiniBand

-----------------------------

## Details of the problem

The following command no longer works after the commit https://github.com/openpmix/openpmix/commit/22dabb8e471fb110ca75a5fd3d5f12d8d6a984a3
```console
$ mpirun -- ./a.out
--------------------------------------------------------------------------
An unrecognized option was included on the mpiexec command line:

  Option: --

Please use the "mpiexec --help" command to obtain a list of all
supported options.
--------------------------------------------------------------------------
```

I often use the double-dash delimiter `--` to explicitly separate command-line options for MPI and commands that I want to execute.
Not sure if this style was officially supported, but I think it was working in prior versions.

**Comment by rhc54** [comment]
It would be more accurate to say "it was fixed, but appears to have problems again". Fix is [here](https://github.com/openpmix/openpmix/pull/3027)

**Comment by s417-lama** [comment]
@rhc54 I appreciate your work all the time, but this issue is happening again with Open MPI v5.0.0. Is there no testing for it?
This time, no error message is shown with `--` but it silently exits without executing commands.

**Comment by rhc54** [comment]
> Is there no testing for it?

Not that I know of, though it could be the OMPI folks do it - though I would then have expected them to report it.

I can take a look in a bit - might be awhile before a fix is issued, though. Is it present on OMPI main as well? Just wondering if something didn't make it across the branches.

**Comment by rhc54** [comment]
Well, FWIW this has nothing to do with PMIx - problem is in PRRTE. Looks like it has been there for a long time but masked by changes in PMIx

**Comment by rhc54** [comment]
This has been fixed by #3199 and https://github.com/openpmix/prrte/pull/1855
